.github/workflows/simulator.yml

```diff
@@ -22,3 +22,15 @@ jobs:
     - name: Test with pytest
       run: |
         python3 -m pytest -vrf
+    - name: Test filter outputs
```

Comment:
You only need lines 33-36 here. Also you should remove the simulator tests from pytest